.github/workflows/basic.yml

```diff
@@ -0,0 +1,101 @@
+name: Basic build and test workflow
+
+on:
+  push:
+    branches: [ "main", "dev" ]
+  pull_request:
+    branches: [ "main", "dev" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_and_test:
+    name: Compile and test
+
+    runs-on: ubuntu-latest
```

Comment:
I enabled ubicloud for this. You can use 'ubicloud-standard-4' when running this action.